### DISCUSSION TITLE
Best way to upgrade HA core from python 3.10 to 3.11?

### DISCUSSION THREAD

**Discussion by dmshimself** [tteck/Proxmox#1500] [General]
I have an older HA core currently running 2023.5.3 and when I tried to upgrade to 2023.6 I get an illegal instruction when HA starts.  Well there have been warnings about moving to python 3.11 for some time and indeed there is a note in the release notes for 2023.6 along these lines.  The old LXC uses python 3.10.  So I thought it would be a good step to upgrade python first and I have tried to find a way to reliably upgrade to 3.11.  But I've lost my way slightly and I'm looking for advice.  I used the 'usual' method using update-alternatives, but there seem to python 3.9 versions installed, 3.10 installed in root/.shim and I couldn't seem to get 3.11 installed system wide.  (I still get the illegal instruction when starting HASS)

So is there a recommended way to update to python 3.11 in the current LXC or would it be best to create a new HA LXC core which I see already uses python 3.11?  I've not installed anything into the old HA Core LXC, so if a new LXC is the way to go, will transferring the contents of a backup from the old LXC into the new LXC config likely be enough to get me back to where I was?

**Comment by tteck** [comment]
https://github.com/tteck/Proxmox/issues/1469

**Comment by dmshimself** [comment]
My old eye let me down.  Many thanks for the pointer

**Reply by tteck** [reply]
You can now use Advanced Settings to utilize Debian 12 for the install
